src/scripting/Scripting.cpp

```diff
@@ -97,13 +97,44 @@ void Scripting::Initialize()
 
     luaGlobal["ModArchiveExists"] = [this](const std::string& acArchiveName) -> bool
     {
+        bool archiveExists = false;
+
         const auto cAbsPath = absolute(m_paths.ArchiveModsRoot() / acArchiveName);
         const auto cRelPathStr = relative(cAbsPath, m_paths.ArchiveModsRoot()).native();
 
         if (cRelPathStr.find(L"..") != std::wstring::npos)
-            return false;
+            archiveExists = false;
+        else
+            archiveExists = exists(cAbsPath);
+
+        // only need to check if it wasn't already found
+        if (!archiveExists)
+        {
+            // check to see if a REDmod archive exists
+            const auto cREDModAbsPathRoot = absolute(m_paths.REDmodsRoot());
+
+            // parse recursively to find the archive
+            for (auto dir_iter = std::filesystem::recursive_directory_iterator{cREDModAbsPathRoot};
```

Comment:
I would not use recursive_directory_iterator when you care about specific level only. 
Also not a fan of this type of iteration this looks better IMO.
Lastly, as mentioned before, use GetLuaPath function for validating things please.
```
for (const auto& redModIter : std::filesystem::directory_iterator(cREDModsAbsPathRoot))
{
    if (!redModIter.is_directory())
        continue;
        
    for (const auto& entry : std::filesystem::directory_iterator(redModIter.path()))
    {
        if (!entry.is_directory())
            continue;
            
        // check path, again using GetLuaPath
    }
}
```

Reply:
no longer valid due to RED4ext::ResourceDepot implementation